src/parser/datetime.rs

```diff
@@ -68,6 +70,92 @@ pub(crate) fn tokenize_interval(value: &str) -> Result<Vec<IntervalToken>, Parse
     Ok(toks)
 }
 
+fn tokenize_timezone(value: &str) -> Result<Vec<IntervalToken>, ParserError> {
+    let mut toks: Vec<IntervalToken> = vec![];
+    let mut num_buf = String::with_capacity(4);
+    /* If the timezone string has a colon, we need to parse all numbers naively, otherwise we need to parse long sequences of digits as [..hhhhmm] */
+    let split_nums: bool = value.find(':').unwrap_or(0) == 0;
+
+    fn parse_num(
+        toks: &mut Vec<IntervalToken>,
+        n: &str,
+        split_nums: bool,
+        idx: usize,
+    ) -> Result<(), ParserError> {
+        if n.is_empty() {
+            return Ok(());
+        }
+
+        if n.len() <= 2 || split_nums == false {
+            let token = IntervalToken::Num(n.parse().map_err(|e| {
+                ParserError::ParserError(format!(
+                    "Unable to parse value as a number at index {}: {}",
+                    idx, e
```

Comment:
this seems like it could include `value` in the output so that you know what the index is into.

Reply:
it didn't have that in the parse_num function in tokenize_interval so I followed that precedent -- if you feel strongly about it I'm happy to include the string in the output

went ahead and added the original string in the output everywhere else

Reply:
I would consider that more of a lack of debuggability in `tokenize_interval` than necessarily something that should be followed, but it doesn't really matter.